### DISCUSSION TITLE
How to pass input parameter to trigger dagruns ?

### DISCUSSION THREAD

**Discussion by enniomorricone123** [apache/airflow#15647] [General]
I have defined a DAG with a couple of tasks using AwsBatchOperator. The function I would like to achieve is that user can trigger this dag with different input ( let's call it dataset_name, this can be different depending on user's input). Inside the dag, I need to use the dataset name to define my task (using awsbatchoperator). 
Here is a snippet of my code:
```
dag = DAG(dag_id='job1', schedule_interval= None, default_args=default_args, catchup=False)

task1 = AwsBatchOperator(
        task_id='task_1",
        job_name="task_1_" + **dataset_name**
        job_queue='queue1",
        job_definition='arn:aws:batch:xxxxx.xxxxx',
        max_retries=200,
        aws_conn_id=None,
        array_properties={},
        parameters={"dataset_name":**dataset_name**},
        overrides={},
        status_retries=10,
        region_name='xxxxx',
        dag=dag)
```

I tried to use {{dag_run.conf }}, but  apparently it is not a template field awsBatchOperator, so I cannot use it. Is there a way to solve this problem?

**Comment by josh-fell** [comment]
Hi @enniomorricone123,

How is the DAG being triggered? What Airflow version are you running?

Both `job_name` and `parameters` arguments can be templated for the `AwsBatchOperator` so you should be able to build the value assuming you include dataset_name in the `conf` object when triggering this DAG:

```python
task1 = AwsBatchOperator(
        task_id='task_1",
        job_name="task_1_" + "{{ dag_run.conf.dataset_name }}",
        job_queue='queue1",
        job_definition='arn:aws:batch:xxxxx.xxxxx',
        max_retries=200,
        aws_conn_id=None,
        array_properties={},
        parameters={"dataset_name":"{{ dag_run.conf.dataset_name }}"},
        overrides={},
        status_retries=10,
        region_name='xxxxx',
        dag=dag
)

**Comment by 2stash** [comment]
If you just want to change the dataset, and run the exact same query/function then @josh-fell 's answer works.

If not, i would try using "BranchPythonOperator", which calls a method. In the method you can put logic to determine what operator should be ran next.
